bitwarden_license/src/Commercial.Core/AdminConsole/Services/ProviderService.cs

```diff
@@ -393,6 +397,94 @@ public async Task AddOrganizationsToReseller(Guid providerId, IEnumerable<Guid>
+        {
+            return;
+        }
+
+        foreach (var orgId in organizationIds)
+        {
+            var organization = await _organizationRepository.GetByIdAsync(orgId);
+            var subscriptionItem = await GetSubscriptionItemAsync(organization.GatewaySubscriptionId, GetPlanId(organization.PlanType));
+            var newPlanType = GetPlanTypeFromPlan(organization.Plan, organization);
+            if (subscriptionItem != null)
+            {
+                await UpdateSubscriptionAsync(organization.GatewaySubscriptionId, subscriptionItem, GetPlanId(newPlanType));
+            }
+
+            await _organizationRepository.UpsertAsync(organization);
+        }
+    }
+
+    private async Task<Stripe.SubscriptionItem> GetSubscriptionItemAsync(string subscriptionId, string oldPlanId)
+    {
+        var subscrptionInfo = await _stripeAdapter.SubscriptionGetAsync(subscriptionId);
+        return subscrptionInfo.Items.Data.FirstOrDefault(item => item.Price.Id == oldPlanId);
+    }
+
+    private static string GetPlanId(PlanType planType)
+    {
+        return StaticStore.GetPlan(planType).PasswordManager.StripeSeatPlanId;
+    }
+
+    private async Task UpdateSubscriptionAsync(string subscriptionId, Stripe.SubscriptionItem subscriptionItem, string newPlanId)
+    {
+        try
+        {
+            if (subscriptionItem.Plan.Id != newPlanId)
+            {
+                await _stripeAdapter.SubscriptionUpdateAsync(subscriptionId,
+                    new Stripe.SubscriptionUpdateOptions
+                    {
+                        Items = new List<Stripe.SubscriptionItemOptions>
+                        {
+                            new Stripe.SubscriptionItemOptions { Id = subscriptionItem.Id, Price = newPlanId },
+                        }
+                    });
+            }
+        }
+        catch (Exception e)
+        {
+            throw new Exception("Unable to update existing plan on stripe");
+        }
+
+    }
+
+    private static PlanType GetPlanTypeFromPlan(string plan, Organization organization)
+    {
+        var planTypeMappings = new Dictionary<PlanType, string>
+        {
+            { PlanType.EnterpriseAnnually2020, GetEnumDisplayName(PlanType.EnterpriseAnnually2020) },
+            { PlanType.EnterpriseMonthly2020, GetEnumDisplayName(PlanType.EnterpriseMonthly2020) },
+            { PlanType.TeamsMonthly2020, GetEnumDisplayName(PlanType.TeamsMonthly2020) },
+            { PlanType.TeamsAnnually2020, GetEnumDisplayName(PlanType.TeamsAnnually2020) }
+        };
+
+        foreach (var mapping in planTypeMappings)
+        {
+            if (mapping.Value.IndexOf(organization.Plan, StringComparison.Ordinal) != -1)
+            {
+                organization.PlanType = mapping.Key;
+                organization.Plan = mapping.Value;
+                return organization.PlanType;
+            }
+        }
+
+        throw new BadRequestException("Invalid PlanType selected");
```

Comment:
This should throw an `ArgumentException` instead of a `BadRequestException` since at this stage we're dealing with our own pre-existing data and not something from a user request.

https://learn.microsoft.com/en-us/dotnet/api/system.argumentexception?view=net-8.0